File: src/main/java/app/entity/Guide.java
package app.entity;

import org.hibernate.annotations.ColumnDefault;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Guide {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	@NotBlank(message = "Titulo nao pode ficar em branco")
	private String titulo;
	
	@NotBlank(message = "Tipo nao pode ficar em branco")
	private String tipo;
	
	@NotBlank(message = "Descricao nao pode ficar em branco")
	private String descricao;
	
	@NotBlank(message = "Link nao pode ficar em branco")
	private String link;
	
	@NotBlank(message = "Data nao pode ficar em branco")
	private String data_cr;
	
	@ColumnDefault("0")
	private Integer likes;
	
	@ColumnDefault("0")
	private Integer dislikes;
	
	@ManyToOne
	@JsonIgnoreProperties({"guides","game"})
	@NotNull(message = "Voce deve selecionar um personagem")
	private Fighter fighter;
	
	@ManyToOne
	@JsonIgnoreProperties({"guides", "fighter", "event"})
	@NotNull(message = "Voce deve selecionar um jogo")
	private Game game;
	
	@ManyToOne
	@JsonIgnoreProperties({"guides", "event"})
	@NotNull(message = "Um usuario deve existir")
	private User user;
}

File: frontend/src/app/model/guide.ts
import { Fighter } from "./fighter";
import { Game } from "./game";
import { User } from "./user";

export class Guide {
  id!: number;
  titulo!: string;
  tipo!: string;
  descricao!: string;
  link!: string;
  data_cr!: Date;
  likes: number = 0;
  dislikes: number = 0;
  fighter!: Fighter;
  game!: Game;
  user!: User;

  constructor(
      titulo: string,
      tipo: string,
      descricao: string,
      link: string,
      fighter: Fighter,
      game: Game,
      user: User,
      data_cr: Date
  ) {
      this.titulo = titulo;
      this.tipo = tipo;
      this.descricao = descricao;
      this.link = link;
      this.fighter = fighter;
      this.game = game;
      this.user = user;
  }
}
